core/src/utils/animation/animation-utils.ts

```diff
@@ -0,0 +1,83 @@
+export const animationEnd = (el: HTMLElement | null, callback: (ev?: TransitionEvent) => void) => {
+  let unRegTrans: (() => void) | undefined;
+  const opts: any = { passive: true };
+
+  const unregister = () => {
+    if (unRegTrans) {
+      unRegTrans();
+    }
+  };
+
+  const onTransitionEnd = (ev: Event) => {
+    if (el === ev.target) {
+      unregister();
+      callback(ev as TransitionEvent);
+    }
+  };
+
+  if (el) {
+    el.addEventListener('webkitAnimationEnd', onTransitionEnd, opts);
+    el.addEventListener('animationend', onTransitionEnd, opts);
+
+    unRegTrans = () => {
+      el.removeEventListener('webkitAnimationEnd', onTransitionEnd, opts);
+      el.removeEventListener('animationend', onTransitionEnd, opts);
+    };
+  }
+
+  return unregister;
+};
+
+export const supportsWebAnimations = (): boolean => {
```

Comment:
since it's const, we could do:
```
export const supportsWebAnimations = /*@__PURE__*/(() => {
return !!(window as any).Animation;
})();
```
so only runs once